code/game/gamemodes/malfunction/newmalf_ability_trees/tree_synthetic.dm

```diff
@@ -0,0 +1,363 @@
+// SYNTHETIC TREE
+//
+// Abilities in this tree allow the AI to upgrade their robotic companions.
+// T1 - Reset Cyborg Module - Allows the AI to reset a slaved cyborgs module.
+// T2 - Infect APC - Gives the AI the ability to infect APC's which can slave IPC's that charge off of them to the AI.
+// T3 - Overclock Cyborg - Allows the AI to give the option to a slaved borg to overclock which increases preformance of the borg.
+// T4 - Synthetic Takeover - Allows the AI to start a station wide takeover based on synthetic dominance.
+
+
+// BEGIN RESEARCH DATUMS
+
+/datum/malf_research_ability/synthetic/reset_module
+	ability = new/datum/game_mode/malfunction/verb/reset_module()
+	price = 100
+	next = new/datum/malf_research_ability/synthetic/infect_apc()
+	name = "Reset Cyborg Module"
+
+
+/datum/malf_research_ability/synthetic/infect_apc
+	ability = new/datum/game_mode/malfunction/verb/infect_apc()
+	price = 700
+	next = new/datum/malf_research_ability/synthetic/overclock_borg()
+	name = "Infect APC"
+
+
+/datum/malf_research_ability/synthetic/overclock_borg
+	ability = new/datum/game_mode/malfunction/verb/overclock_borg()
+	price = 1300
+	next = new/datum/malf_research_ability/synthetic/synthetic_takeover
+	name = "Overclock Cyborg"
+
+
+/datum/malf_research_ability/synthetic/synthetic_takeover
+	ability = new/datum/game_mode/malfunction/verb/synthetic_takeover()
+	price = 5000
+	name = "Synthetic Takeover"
+
+// END RESEARCH DATUMS
+// BEGIN ABILITY VERBS
+
+/datum/game_mode/malfunction/verb/reset_module(var/mob/living/silicon/robot/target = null as mob in get_linked_cyborgs(usr))
+	set name = "Reset Cyborg Module"
+	set desc = "100 CPU - Forces a module reset on a cyborg that is enslaved to you. Has a chance of failing."
+	set category = "Software"
+	var/price = 100
+	var/mob/living/silicon/ai/user = usr
+	if(!ability_prechecks(user, price))
+		return
+
+	if(target && !istype(target))
+		user << "This is not a cyborg."
+		return
+
+	if(target && target.connected_ai && (target.connected_ai != user))
+		user << "This cyborg is not connected to you."
+		return
+
+	if(!target)
+		var/list/robots = list()
+		var/list/robot_names = list()
+		for(var/mob/living/silicon/robot/R in world)

```

Comment:
Nooooo.

Use the mobs global list? Unless there's a silicons global list. Check global_lists.dm.